src/python/WMCore/WorkQueue/WMBSHelper.py

```diff
@@ -737,10 +737,10 @@ def validFiles(self, files):
                 continue
 
             # Create a LumiList from the WMBS info
-            lumis = []
+            runLumis = {}
             for x in f['LumiList']:
-                lumis.append([str(x['RunNumber']), x['LumiSectionNumber']])
-            fileLumiList = LumiList(lumis=lumis)
+                runLumis[str(x['RunNumber'])] = x['LumiSectionNumber']
```

Comment:
Lumi list should now sort things. If not, I will make it do so. But yeah, you will want to do a conditional extend.

I have not looked at why some tests are still failing.

Eric